src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -140,6 +140,34 @@
  * </li>
  * </ul>
  * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name="ImportOrder"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * import java.io.IOException; // Ok
+ * import java.net.URL; // Ok
+ *
+ * import java.io.IOException; // Violation, Extra separation and wrong order
```

Comment:
`Extra separation and wrong order`
1. Please add clarification where this should be  (wrong order)
2. It is not clear where is extra separation - from top or bottom?
3. Please make violation description lowercase here and below (Extra -> extra)

Reply:
i wanted to add but i was getting some error like line length exceed 100
so how should i describe both violation in line 155?

Reply:
> It is not clear where is extra separation - from top or bottom?

it's always from the top
i guess!

Reply:
> i wanted to add but i was getting some error like line length exceed 100
so how should i describe both violation in line 155?

you can wrap comment, like https://checkstyle.org/config_naming.html#CatchParameterName_Examples

>we have config and above config some notes

It is not needed to write a lot, just like `should be in java.io group` or something like this is enough

Reply:
```
} catch (Exception FourthException) { // violation, the initial letter
                                          // should be lowercase
      // ...
    }
```
`// ...` this will count as line break

Reply:
> you can wrap comment

this is giving different CLI output (extra separation violation)

Reply:
i have a suggestion- instead of writing extra separation or wrong order we can write the property name like
//violation, of property separated and ordered.

Reply:
> //violation, of property separated and ordered.

It is better not be cryptic and write it as clear as possible. Message like `// violation, should be in group X` is clearer than mentioning properties

> this is giving different CLI output (extra separation violation)

I think it is fine to have an extra violation if description will be more useful for users

Reply:
i have a doubt 
ImportOrder [Description](https://checkstyle.org/config_imports.html#ImportOrder_Description)
check's 1st example should be according to description or default values of property from property table?

Reply:
also some changes in this example

```
import java.io.IOException; // OK
import java.net.URL; // OK

import java.io.IOException; // violation, extra separation before 'java.io.IOException'.
                            // and wrong order for 'java.io.IOException' import.
import javax.swing.JComponent; // violation, extra separation before 'javax.net.ssl.SSLContext'
import javax.net.ssl.TrustManager;
import org.apache.http.conn.ClientConnectionManager; // OK
import java.util.Set; // violation, wrong order for 'java.util.Set' import.
import com.neurologic.http.HttpClient; // violation, wrong order for import
import com.neurologic.http.impl.ApacheHttpClient; // OK

public class SomeClass { }
```

Reply:
>check's 1st example should be according to description or default values of property from property table?

It should show the default config. Yes, it means default values for all properties.

Reply:
so this example is ok or should i change anything?

Reply:
ok so let me make such example and if u like it we will switch it with the current one. okay?